### DISCUSSION TITLE
No objects on listing

### DISCUSSION THREAD

**Discussion by neha04** [pimcore/pimcore#16415] [Q&A / Support]
Hi all,

i am using pimcore x
i have created class product and when i am trying to list all objects , i am getting null array.
Objects of that class exists in the system.
never faced such issue before.

**Comment by mlkdev** [comment]
Ensure your objects are published. Listings silently require publish status as true. If you'd like to return un-published objects try something like this:

```
<?php

use Pimcore\Model\DataObject\MyObject;
...
$Listing = new MyObject\Listing();
$Listing->setUnpublished( true );
```

Just remember to replace the "MyObject" with the name of your object's class.

Here's a link to the core docs:
https://pimcore.com/docs/platform/Pimcore/Objects/Working_with_PHP_API/#get-an-object-list-including-unpublished-objects
